apps/tx-builder/src/scripts/deploy-test-matrix-contract.js

```diff
@@ -0,0 +1,36 @@
+const RINKEBY_ETHERSCAN_URL = 'https://rinkeby.etherscan.io'
+const TEST_CONTRACT_NAME = 'MatrixTypesTestContract'
+
+async function main() {
+  // A ContractFactory in ethers.js is an abstraction used to deploy new smart contracts,
+  const TestContract = await ethers.getContractFactory(TEST_CONTRACT_NAME)
+
+  console.log('Deploying Test Contract...')
+  console.log('')
+
+  // Start deployment, returning a promise that resolves to a contract object
+  const { address: testContractAddress } = await TestContract.deploy()
+
+  console.log('New Matrix Test Contract deployed in Rinkeby')
+  console.log('')
+
+  console.log('Matrix Test Contract Address: ', testContractAddress)
+  console.log('')
+
+  const etherscanUrl = `${RINKEBY_ETHERSCAN_URL}/address/${testContractAddress}`
+  console.log('Etherscan URL: ', etherscanUrl)
+  console.log('')
+
+  const verifyCommand = `yarn contract:verify:rinkeby ${testContractAddress}`
```

Comment:
you can also compose deployment & verification into hardhat tasks like in safe-contracts: https://github.com/safe-global/safe-contracts/blob/main/src/tasks/deploy_contracts.ts

Reply:
nice! makes sense! :3

Reply:
I updated the hardhat configuration with your suggestion and now its the Final Fantasy! 

just with this command:

```
yarn contract:deploy-all rinkeby
```
or
```
yarn contract:deploy-all xdai
```
both contract are deployed and verified!

![Captura de pantalla 2022-06-08 a las 14 20 35](https://user-images.githubusercontent.com/26763673/172614853-02f872f3-235f-46d5-9068-2e1835de99c9.png)


in `gnosis chain` the command shows and error but the contracts gets verified and the ABI is automatically fetched from sourcify!!!! :D 

![Captura de pantalla 2022-06-08 a las 14 21 10](https://user-images.githubusercontent.com/26763673/172614869-b95b956f-8b0e-4798-8c1c-3c270be820ea.png)


Thanks! :3